src/FluentValidation.AspNetCore/FluentValidationModelValidatorProvider.cs

```diff
@@ -81,23 +81,30 @@ public class FluentValidationModelValidator : IModelValidator {
 				var interceptor = customizations.GetInterceptor()
 				                  ?? validator as IValidatorInterceptor
 				                  ?? mvContext.ActionContext.HttpContext.RequestServices.GetService<IValidatorInterceptor>();
+				var actionContextInterceptor = customizations.GetActionContextInterceptor()
+				                  ?? validator as IActionContextValidatorInterceptor
+				                  ?? mvContext.ActionContext.HttpContext.RequestServices.GetService<IActionContextValidatorInterceptor>();
 
 				IValidationContext context = new ValidationContext<object>(mvContext.Model, new PropertyChain(), selector);
 				context.RootContextData["InvokedByMvc"] = true;
 				context.SetServiceProvider(mvContext.ActionContext.HttpContext.RequestServices);
 
-				if (interceptor != null) {
+				if (interceptor != null && mvContext.ActionContext is ControllerContext) {
 					// Allow the user to provide a customized context
 					// However, if they return null then just use the original context.
 					context = interceptor.BeforeMvcValidation((ControllerContext)mvContext.ActionContext, context) ?? context;
+				} else if (actionContextInterceptor != null) {

```

Comment:
Let me know if you think the logic makes sense.

In the event there are two interceptors registered, the existing one takes precedence, unless the `ActionContext` is not a `ControllerContext` then the other interceptor will pick up.

Reply:
Looks good